// brunch-extensions/lib/languages/base.js
(function() {
  var async, fs, pathModule,
    __slice = Array.prototype.slice;

  fs = require('fs');

  pathModule = require('path');

  async = require('async');

  exports.BaseLanguage = (function() {

    BaseLanguage.prototype.queue = async.queue(fs.readFile, 5);

    function BaseLanguage(config) {
      this.config = config;
    }

    BaseLanguage.prototype.getRootPath = function() {
      var subPathes;
      subPathes = 1 <= arguments.length ? __slice.call(arguments, 0) : [];
      return pathModule.join.apply(pathModule, [this.config.rootPath].concat(__slice.call(subPathes)));
    };

    BaseLanguage.prototype.getBuildPath = function() {
      var subPathes;
      subPathes = 1 <= arguments.length ? __slice.call(arguments, 0) : [];
      return pathModule.join.apply(pathModule, [this.config.buildPath].concat(__slice.call(subPathes)));
    };

    BaseLanguage.prototype.readFile = function(path, callback) {
      return this.queue.push(path, function(error, data) {
        return callback(error, data.toString());
      });
    };

    BaseLanguage.prototype.compile = function(path, callback) {
      return this.readFile(path, callback);
    };

    return BaseLanguage;

  })();

}).call(this);
